### PR TITLE
Use hyperopt to find optimal parameters for buy strategy

### PR DESCRIPTION
**Pull request by vertti**
You need to install:

```
pip install hyperopt
pip install networkx==1.11
```
(second dependency is due to a current bug in hyperopt).

Then adjust the number of times you want to run the backtesting in search of the best parameter values. Open `test_hyperopt.py` and change the `max_evals` on the last line:

```
    print(fmin(fn=optimizer, space=space, algo=tpe.suggest, max_evals=2))
```

then run the test with:
```
$ BACKTEST=true pytest -s freqtrade/tests/test_hyperopt.py
```

### CODE REVIEW

**Comment by gcarq** [comment]
Do we no longer need `setup.cfg`? Looks good to me otherwise. Good job :-)

**Comment by vertti** [comment]
Well, we use the direct command from setup, rather than the "test" alias and it's parameters from `setup.cfg` so I don't think we need it. Plus now pytest will always pick it's settings from setup.cfg even when just running it from command line. That will cause it to always do full coverage and testsuite which is super annoying when you just want to run hyperopt or backtesting.

File: freqtrade/tests/test_hyperopt.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+            {'enabled': True}
+        ]),
+        'trigger': hp.choice('trigger', [
+            {'type': 'lower_bb'},
+            {'type': 'faststoch10'}
+        ]),
+    }
+
+    # print(hyperopt.pyll.stochastic.sample(space))
+    print('Best parameters {}'.format(fmin(fn=optimizer, space=space, algo=tpe.suggest, max_evals=40)))
```
Comment (shusso) [inline freqtrade/tests/test_hyperopt.py]: Should the `max_evals` be a "bit" smaller by default. In my machine only `10` evals takes ~190s.

**Comment by shusso** [comment]
Also now the backtest code is duplicated in `test_hyperopt` and in `test_backtesting`. @vertti could something like [this](https://github.com/shusso/freqtrade/commit/210d5e83f5acd8fdf541aa1b9613ee5b82a533b7) work?

File: freqtrade/tests/test_hyperopt.py (hunk 1 above)
Comment (vertti) [reply freqtrade/tests/test_hyperopt.py]: Well the test is also disabled by default. You need `BACKTEST=true` to run it.

**Comment by vertti** [comment]
@shusso yeah, that combining looks good. I'd vote for merging this in and then you can make a PR out of that refactoring.

**Pull request merged**